org.civicrm.reqangular/test/services/check-perm_test.js

```diff
@@ -0,0 +1,49 @@
+(function (CRM) {
+  define([
+    'common/angular',
+    'common/angularMocks',
+    'common/services/check-perm'
+  ], function () {
+    'use strict';
+
+    describe('CheckPermissions', function () {
+      var $q, $rootScope, checkPermissions;
+
+      beforeEach(module('common.services'));
+
+      beforeEach(inject(function (_$rootScope_, _$q_, _CheckPermissions_) {
+        $rootScope = _$rootScope_;
+        $q = _$q_;
+        checkPermissions = _CheckPermissions_;
+      }));
+
+      describe('canAdmin', function () {
+        var promise;
+
+        beforeEach(function () {
+          spyOn(CRM, 'checkPerm').and.callFake(function () {
```

Comment:
We dont need a promise here, `checkPerm` does not return promise.
`spyOn(CRM, 'checkPerm').and.callFake(returnTrueIfPermissionIs('CiviHRLeaveAndAbsences: Administer Leave and Absences'));` would still work.